docs/tutorial/typesfuns.rst

```diff
@@ -661,6 +661,93 @@ without any explicit use of ``Force`` or ``Delay``:
     ifThenElse True  t e = t;
     ifThenElse False t e = e;
 
+Codata Types
+============
+
+Codata types are like regular data types, except that they allow for us to
+define infinite data structures.  More precisely, for a type ``T``, each of its
+constructor arguments of type ``T`` are transformed into a coinductive parameter
+``Inf T``. This makes each of the ``T`` arguments lazy, and allows infinite data
+structures of type ``T`` to be built. One example of a codata type is Stream,
+which is defined as follows.
+
+.. code-block:: idris
+
+    codata Stream : Type -> Type where
+      (::) : (e : a) -> Stream a -> Stream a
+
+This gets translated into the following by the compiler.
+
+.. code-block:: idris
+
+    codata Stream : Type -> Type where
```

Comment:
I wasn't. (not my example) But you are right. There is a mistake there.

On Fri, Feb 26, 2016 at 12:28 AM, Leif Warner <notifications@github.com>
wrote:

> In docs/tutorial/typesfuns.rst
> <https://github.com/idris-lang/Idris-dev/pull/2988#discussion_r54176198>:
>
> > +define infinite data structures.  More precisely, for a type ``T``, each of its
> > +constructor arguments of type ``T`` are transformed into a coinductive parameter
> > +``Inf T``. This makes each of the ``T`` arguments lazy, and allows infinite data
> > +structures of type ``T`` to be built. One example of a codata type is Stream,
> > +which is defined as follows.
> > +
> > +.. code-block:: idris
> > +
> > +    codata Stream : Type -> Type where
> > +      (::) : (e : a) -> Stream a -> Stream a
> > +
> > +This gets translated into the following by the compiler.
> > +
> > +.. code-block:: idris
> > +
> > +    codata Stream : Type -> Type where
>
> Are you replying to my comment above? I just meant you can leave your
> existing example as-is (which has direct self-recursion), only changing the
> "desugared" form from codata to data (it already has the Inf manually
> inserted on the recursive param).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idris-lang/Idris-dev/pull/2988/files#r54176198>.
>